lib/capistrano/asg/tasks/rolling.rake

```diff
@@ -106,11 +106,16 @@ namespace :rolling do
       end
     end
 
-    instances = config.instances.auto_terminate
-    if instances.any?
-      logger.info 'Terminating instance(s)...'
-      instances.terminate
+    begin
+      instances = config.instances.auto_terminate
+      if instances.any?
+        logger.info 'Terminating instance(s)...'
+        instances.terminate
+      end
+    rescue Aws::Errors::ServiceError => e
+      logger.warning "Error deleting instance: #{e}"
```

Comment:
I'm curious if you did encounter a situation where terminating the instance failed?

Reply:
Yes many times, and rescuing was the only way through this.

Reply:
Hmm interesting, we've never seen that before. Any idea what the error was or what could have caused this?
And did you needed to terminate the instance manually?

Reply:
I'm afraid I don't have a record. It may have been whilst we were in the process of switching to autoscaling groups and would not appear normally :shrug: